### PR TITLE
Remove Batch processing from prompts and update tests

### PR DESCRIPTION
**Pull request by cleemullins**
Removes the batch processing from prompts. This also means the Core.Extensions assembly is no longer needed as a reference, and the BatchMiddleware is not needed in any way. 

Discussed yesterday with Tom & Steve, and Node will update to also removing batching. 

Also tracked in #293

### CODE REVIEW

**Comment by brandonh-msft** [comment]
cc @joescars who flagged this issue. From what I understand he's simply using `TextPrompt` and it's requiring that he register the `BatchOutputMiddleware`

File: libraries/Microsoft.Bot.Builder.Prompts/BasePrompt.cs [hunk 1]
```diff
@@ -76,7 +75,8 @@ public Task Prompt(IBotContext context, string text, string speak = null)
         /// </summary>
         public Task Prompt(IBotContext context, IMessageActivity activity)
         {
-            context.Batch().Reply(activity);
+            // Note: Cast can be removed once PR #284 is merged. Currently in CR. 
+            context.SendActivity((Activity)activity);
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Prompts/BasePrompt.cs]: > context.SendActi [](start = 11, length = 17)

should be 
```
return context.SendActivity();
```

File: tests/Microsoft.Bot.Builder.Prompts.Tests/TemperaturePromptTests.cs [hunk 2]
```diff
@@ -39,10 +39,10 @@ public class TemperaturePromptTests
                             Assert.IsNotNull(tempResult.Text);
                             Assert.IsNotNull(tempResult.Unit);
                             Assert.IsInstanceOfType(tempResult.Value, typeof(float));
-                            context.Batch().Reply($"{tempResult.Value} {tempResult.Unit}");
+                            await context.SendActivity($"{tempResult.Value} {tempResult.Unit}");
```
Comment (tomlm) [inline tests/Microsoft.Bot.Builder.Prompts.Tests/TemperaturePromptTests.cs]: >SendActivity [](start = 42, length = 12)

This naming doesn't seem right
SendActivity() implies that we know who to send it to.  

I think the naming should be
```
Task<ResourceResponse> SendActivity(Activity)  
Task<ResourceResponse> Reply(text, ssml);
```
The point being SendActivity is fully routed activity (including proactive cases) and Reply() is specifically a reply to sender

Comment (cleemullins) [reply tests/Microsoft.Bot.Builder.Prompts.Tests/TemperaturePromptTests.cs]: Per our discussion, this is a bigger picture issue we need to resolve across frameworks, and we're punting on this for today.

**Pull request merged**